src/mongocxx/test_util/client_helpers.cpp

```diff
@@ -50,6 +50,19 @@ using namespace bsoncxx;
 using bsoncxx::builder::basic::kvp;
 using bsoncxx::builder::basic::make_document;
 
+namespace {
+// These frequently used network calls are cached to avoid bottlenecks during tests.
+document::value get_is_master(const client& client) {
+    static auto reply = client["admin"].run_command(make_document(kvp("isMaster", 1)));
+    return reply;
+}
+
+document::value get_server_status(const client& client) {
+    static auto status = client["test"].run_command(make_document(kvp("serverStatus", 1)));
```

Comment:
Aside: Using the `admin` database is necessary for some administrative commands, like `currentOp`. The database used here for `serverStatus` or `isMaster` could be anything.

Reply:
Gotcha. Should I change this to "admin" for consistency? I'm not sure if that'd break anything (this was only ever called with the "test" database) but I can change it and let the test re-run to find out.

Reply:
It should be equivalent and safe to change to `admin`. The C test runner `call_ismaster` uses admin. It makes no behavioral difference, so I am fine either way.

Reply:
Ok. Done. Waiting for the tests to pass then merging.